pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java

```diff
@@ -162,10 +162,10 @@ public void init(PinotConfiguration brokerConf)
     _zkServers = brokerConf.getProperty(Helix.CONFIG_OF_ZOOKEEPR_SERVER).replaceAll("\\s+", "");
     _clusterName = brokerConf.getProperty(Helix.CONFIG_OF_CLUSTER_NAME);
     ServiceStartableUtils.applyClusterConfig(_brokerConf, _zkServers, _clusterName, ServiceRole.BROKER);
+    applyCustomConfigs(brokerConf);
```

Comment:
Shouldn't this be called at the end of init when the rest of the config setting is completed?

Reply:
We want the custom starter being able to add configs to be picked up in the following initialization code. If custom starter wants to add other logic after `init()`, right now it can override `init()` and add code after `super.init()`. We can also consider adding a hook called `postInit()`